duplex_http_call.go

```diff
@@ -168,6 +168,8 @@ func (d *duplexHTTPCall) CloseWrite() error {
 	// response to read from.
 	if d.requestSent.CompareAndSwap(false, true) {
 		go d.makeRequest()
+		// We never setup a request body, so it's effectively already closed.
+		// So nothing else to do.
```

Comment:
Unrelated, but I was reading this code and confused at first why it does nothing to close the body. And it's because we haven't set one yet: if `d.requestSent` is false then the request body is still `http.NoBody`.